controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -845,11 +852,22 @@ func (r *BareMetalHostReconciler) actionProvisioning(prov provisioner.Provisione
 	}
 
 	// If the provisioner had no work, ensure the image settings match.
-	if info.host.Status.Provisioning.Image != *(info.host.Spec.Image) {
+	if info.host.Spec.Image == nil && info.host.Status.Provisioning.Image.URL != "" {
```

Comment:
How can this happen - shouldn't the Status.Provisioning.Image always be removed by deprovisioning?

If this is a corner case found writing the test, I'd break it out into a separate commit as I'm currently not seeing how it's related to this change?

Reply:
> shouldn't the Status.Provisioning.Image always be removed by deprovisioning?

This is provisioning code. Previously we would never get here if Spec.Image is nil, now we can. The old code causes segfault if we use Image == nil and CustomDeploy != nil.

Reply:
Again perhaps I'm missing something, can you explain in more detail why we can now hit the provisioning path with an image set in the status?  IIUC that should never happen, because we should have deprovisioned first.  /cc @zaneb

Reply:
I'm not sure why you're asking me, that's the code that existed before my change. I just made sure it doesn't fail with segfault now that Spec.Image can be niI.

I assume it's this code that synchronizes Status with Spec after deployment, but I may be wrong.

Reply:
Looking more at the code. I can remove this check if you think it's safe to, but I'm not 100% sure myself. Previously this code path would never be hit with Spec.Image == nil.

Reply:
I'm asking you because the `info.host.Status.Provisioning.Image = metal3v1alpha1.Image{}` is new - and AFAIK we should never clear the image in the status during provisioning, only during deprovisioning

So, I'd like to understand under what circumstances the state machine can allow us to get into this state, whether the issue is existing code or not :)

Reply:
To clarify, I'm saying we should only clear the image in [actionDeprovisioning](https://github.com/metal3-io/baremetal-operator/blob/master/controllers/metal3.io/baremetalhost_controller.go#L917) - this addition to `actionProvisioning` seems wrong, so I want to understand if/why it's needed.